File: benchmarks/numprime/implementations/c/numprime.c
#include <math.h>

#include "numprime.h"

int numprime(int n) {
	int total = 0;
	for (int i = 2; i <= n; ++i) {
		int is_prime = 1;
		for (int j = 2; j <= sqrt(i); ++j) {
			if (i % j == 0) {
				is_prime = 0;
				break;
			}
		}
		total += is_prime;
	}
	return total;
}

File: benchmarks/numprime/implementations/matlab/numprime.m
function total = numprime(n)
    total = 0;
    for i = 2:n
        prime = 1;
        for j = 2:sqrt(i)
            if (mod(i,j) == 0)
                prime = 0;
                break;
            end
        end
        total = total + prime;
    end
end
